src/app/reducers/area.reducer.ts

```diff
@@ -5,7 +5,7 @@ import { AreaState, initialState } from './../states/area.state';
 
 export type Action = AreaActions.All;
 
-export const AreaReducer: ActionReducer<AreaState> = (state = initialState, action: Action) => {
+export function AreaReducer(state = initialState, action: Action) {
```

Comment:
I prefer exporting constants as arrow functions so that you can use the 'this' reference if needed anywhere. Well, reducers being pure functions it totally makes sense not to use any external reference hence this change is okay.
Can you add the return types to the functions though? i.e. for this one, it'd be - 
```
export function AreaReducer(state = initialState, action: Action):  ActionReducer<AreaState> {
```

Reply:
I do too but typescript / angular can't figure out what is going on here. The latest version doesn't transpile it correctly. If you want, reset the last 2 commits and you can see the error.

Reply:
`this` will never be used here, so the change is fine however I an error because of the use of a big arrow function is very concerning considering it's a standard language construct.

Reply:
If you can find another way, I would be happy to take it. This is the only thing that allowed me to move forward. Just reset the last 2 commits and try to get it to build.